lib/response.js

```diff
@@ -70,18 +70,43 @@ res.status = function status(code) {
  *      last: 'http://api.example.com/users?page=5'
  *    });
  *
- * @param {Object} links
+ *    res.links([{
+ *          href: 'http://api.example.com/users?page=2',
+ *          rel: 'next',
+ *          title: 'next chapter',
+ *          type: 'text/plain;charset=UTF-8'
+ *      }, {
+ *          href: 'http://api.example.com/users?page=5',
+ *          rel: 'last',
+ *          title: 'last chapter',
+ *          type: 'text/plain;charset=UTF-8'
+ *      }]);
+ *
+ *
+ * @param {Object|Array} links
  * @return {ServerResponse}
  * @public
  */
 
-res.links = function(links){
+res.links = function(links) {
   var link = this.get('Link') || '';
   if (link) link += ', ';
-  return this.set('Link', link + Object.keys(links).map(function(rel){
-    return '<' + links[rel] + '>; rel="' + rel + '"';
-  }).join(', '));
-};
+  if (Array.isArray(links)) { 
+    var attrTest = /^[a-z][a-z-*]+$/;
```

Comment:
This regexp allows for numeric-only attributes to be allowed. Check [this test](https://github.com/strongloop/express/pull/2702/files#diff-c9326b21b33de566e4657c263a2e24b4R105)

There is a bit more conversation about this here on #2619 This PR is a continuation of that, since the original author hasn't responded for a while.